docs(how-to): database instances with private networks (#224)
scaleway/docs-content: storage/database/how-to/connect-database-private-network.mdx

---
title: How to connect a Database Instance to a Private Network
description: This page explains how to connect a Database Instance to a Private Network
dates:
  validation: 2021-11-15
  posted: 2019-11-15
---

[Private Networks](/network/vpc/quickstart/) allows you to configure your own Layer-2 network with IPv4 adresses. Thanks to this feature, you enhance the security of your systems architecture by isolating it from the Internet. 

> This feature comes with a significant change in the architecture of Scaleway's managed Databases: when using Private Network, your application connects directly to your Database nodes, without going through a Load Balancer. 

Reviewer: ```suggestion
This feature comes with a significant change in the architecture of Scaleway's Managed Databases: when using Private Networks, your application connects directly to your Database nodes without going through a Load Balancer. 
```

> This improves performance by reducing the latency between your nodes. It also increases the security of your databases, as Instances in your Private Network can directly to your Database Instance, without passing through the public Internet.

Reviewer 2: I think there's a word missing after "as Instances in your Private Network can directly"

> This unleashes the DB performances by reducing the latency between your nodes and increases the security of your databases, as Instances in your Private Network can directly to your Database Instance, without passing over the public Internet.

Reviewer: ```suggestion
This improves performance by reducing the latency between your nodes. It also increases the security of your databases, as Instances in your Private Network can directly to your Database Instance, without passing through the public Internet.
```

> You can create a new Database Instance on a Private Network or connect an existing Database Instance by adding a new Private Network Endpoint on it. 

Reviewer: ```suggestion
You can create new Database Instances to attach to your Private Network, or attach existing Database Instances by adding Private Networks endpoints to them. 
```

Reviewer: Maybe it would also be good to point out here that (if I've understood correctly) the Private Network can have Instances as well as Databsaes attached to it?

<Message type="requirement">

- You have an account and are logged into the [Scaleway Console](https://console.scaleway.com)
- You have a [Managed Database](/storage/database/concepts/#managed-database/)
- You have [generated an API key](/console/my-project/how-to/generate-api-key/)

</Message>

## How to attach a Database Instance to a Private Network

1. Click **Database** in the **Storage** section of the [Scaleway Console](https://console.scaleway.com) side menu. A list of your Database Instances displays.
2. Click the database name or <Icon name="more" /> > **More info** to access the Database information page.
3. Click the **Private Networks** tabs. The Private Network configuration displays.
4. Click **+ Attach a Private Network**. The Private Network configuration wizard displays.

>   * Choose if you want either to:

Reviewer: ```suggestion
  * Choose whether to:
```

    * Attach an existing Private Network and select the network from the drop-down list.
    * Attach a new Private Network and enter the name of the network in the form. 
  * Enter the static IP for the Database Instance. 
    <Message type="note">

>     You can assign a static IP address of your IP range to your Database Instance. Your static IP must be at least the second ip in the range.

Reviewer: ```suggestion
    You can assign a static IP address from your IP range to your Database Instance. Your static IP must be at least the second IP in the range.
```

---
title: How to connect a Database Instance to a Private Network
description: This page explains how to connect a Database Instance to a Private Network
dates:
  validation: 2021-11-15
  posted: 2019-11-15
---

> [Private Networks](/network/vpc/quickstart/) allows you to configure your own Layer-2 network with IPv4 adresses. Thanks to this feature, you enhance the security of your systems architecture by isolating it from the Internet. 

Reviewer 3: ```suggestion
[Private Networks](/network/vpc/quickstart/) allows you to configure your own Layer-2 network with IPv4 addresses. That way, you can enhance the security of your system's architecture by isolating it from the Internet. 
```

3. Click the **Private Networks** tabs. The Private Network configuration displays.
4. Click **+ Attach a Private Network**. The Private Network configuration wizard displays.
  * Choose whether to:
    * Attach an existing Private Network and select the network from the drop-down list.
    * Attach a new Private Network and enter the name of the network in the form. 
  * Enter the static IP for the Database Instance. 
    <Message type="note">

    You can assign a static IP address from your IP range to your Database Instance. Your static IP must be at least the second IP in the range.

    </Message>

    <Message type="tip">

    You can use the Database virtual IP (VIP) in a DHCP-controlled environnement by [reserving the IP](/network/vpc/how-to/configure-a-public-gateway/#how-to-review-and-configure-dhcp) in the DHCP server configuration to be sure it is not assigned to any other Instance, then configure it on the Private Networks endpoint.

    </Message>
5. Click **Attach a Private Network** to attach the IP address to your Instance. Your static IP is now assigned to the Database Instance: 
  <Lightbox src="scaleway-dbaas-pn-attached.png" alt="" />

<Message type="tip">

> Private Networks are also available on [Virtual Instances](https://www.scaleway.com/en/virtual-instances/). Follow [this guide](/compute/instances/how-to/use-private-networks/) to attach an Instance to a Private Network and configure a persistent IP address on it.

Reviewer 3: same comment as above

Author: ```suggestion
Private Networks are also available on [Virtual Instances](https://www.scaleway.com/en/virtual-instances/). Refer to the page on [how to use Private Networks ](/compute/instances/how-to/use-private-networks/) to attach an Instance to a Private Network and configure a persistent IP address on it.
```

</Message>

## How to detach a Database Instance from a Private Network

1. Click **Database** in the **Storage** section of the [Scaleway Console](https://console.scaleway.com) side menu. A list of your Database Instances displays.
2. Click the database name or <Icon name="more" /> > **More info** to access the Database information page.
3. Click the **Private Networks** tabs. The Private Network configuration displays.
4. Click <Icon name="unlink" /> next to the static IP that you want to unlink from your Database Instance. 

> 4. Click **Detach this Instance** to remove the Instance from your Private Network.

Reviewer 3: ```suggestion
5. Click **Detach this Instance** to remove the Instance from your Private Network.
```
